src/pages/visualization-view/visualization-view.js

```diff
@@ -66,11 +70,20 @@ const VisualizationView = (props) => {
           </span>
           <span className='action-area'>
             <React.Fragment>
-              <TabButton disabled={!isSaveable} className={`header-item save-button ${isDialogOpen && 'saving'}`} onClick={handleSave} >
+              <TabButton disabled={!isSaveable} className={`header-item save-icon ${isDialogOpen && 'saving'}`} onClick={openDialog} >
                 <div title='Save Visualization'><SaveIcon /></div>
               </TabButton>
               <AutoAnchoringPopover className='popover-anchor' open={isDialogOpen} onClose={closeDialog} classes={{ paper: 'popover', root: 'popover-root' }} >
-                <SaveIndicator saving={isSaving} success={isSaveSuccess} failure={isSaveFailure} linkUrl={linkUrl} />
+                {isTitleSet ? 
+                <SaveIndicator saving={isSaving} success={isSaveSuccess} failure={isSaveFailure} linkUrl={linkUrl} /> 
+                :
+                <div>
+                  <b>Query Title: </b> 
+                  <input className="prompt" type="text" name="queryTitle" placeholder="Query Name" value={title} onChange={handleTitleChange} required="true"></input>
+                  <br/>
+                  <button className="save-button" onClick={handleSave} disabled={!isTitleSet}>Save</button>
+                  <button onClick={closeDialog}>Cancel</button>
+                </div>}
```

Comment:
Seems like we should break this little title dialog out into its own component.

Reply:
Talked to Tony and decided the juice was not worth the squeeze.